gateway-service/src/test/java/org/zowe/apiml/gateway/filters/proxyheaders/AdditionalRegistrationGatewayRegistryTest.java

```diff
@@ -0,0 +1,198 @@
+
+    private final InstanceInfo GW1_INSTANCE_INFO = InstanceInfo.Builder.newBuilder()
+        .setAppName(CoreService.GATEWAY.getServiceId())
+        .setInstanceId(GW1_INSTANCE_ID)
+        .setHostName(GW1_HOSTNAME)
+        .setIPAddr(GW1_IP_ADDRESS)
+        .build();
+
+    private final Application GW1_APPLICATION =
+        new Application(CoreService.GATEWAY.getServiceId(),
+            Arrays.asList(GW1_INSTANCE_INFO));
+
+    // Gateway resolved to multiple ip addresses
+    private final String GW2_INSTANCE_ID = "gw2-instance-id";
+    private final String GW2_IP_ADDRESS = "3.3.3.3";
+    private final String GW2_IP_ADDRESS_FROM_DNS = "4.4.4.4";
+    private final String GW2_HOSTNAME = "gw2-hostname";
+
+    private final InstanceInfo GW2_INSTANCE_INFO =
+        InstanceInfo.Builder.newBuilder()
+            .setAppName(CoreService.GATEWAY.getServiceId())
+            .setInstanceId(GW2_INSTANCE_ID)
+            .setHostName(GW2_HOSTNAME)
+            .setIPAddr(GW2_IP_ADDRESS)
+            .build();
+
+    private final Application GW2_APPLICATION =
+        new Application(CoreService.GATEWAY.getServiceId(),
+            Arrays.asList(GW2_INSTANCE_INFO));
+
+    // Appplication with both gateways
+    private Application ALL_GW_APPLICATION =
+        new Application(CoreService.GATEWAY.getServiceId(),
+            Arrays.asList(GW1_INSTANCE_INFO, GW2_INSTANCE_INFO));
+
+    @BeforeEach
+    @SneakyThrows
+    void setup() {
+        discoveryClientMock = Mockito.mock(DiscoveryClient.class);
+        gatewayRegistry = new AdditionalRegistrationGatewayRegistry();
+        gatewayRegistry.registryExpiration = Duration.ofSeconds(300);
+        gatewayRegistry.init();
+    }
+
+    @Test
+    void eventListenerRegistered() {
+        EurekaClientConfig mockEurekaClientConfig = Mockito.mock(EurekaClientConfig.class);
+        when(discoveryClientMock.getEurekaClientConfig()).thenReturn(mockEurekaClientConfig);
+        when(mockEurekaClientConfig.getEurekaServerServiceUrls(any()))
+            .thenReturn(Collections.singletonList("dummy"));
+
+        gatewayRegistry.registerCacheRefreshEventListener(discoveryClientMock);
+
+        verify(discoveryClientMock, times(1))
+            .registerEventListener(any());
+    }
+
+    @Test
+    void whenGateway_thenAddIpToRegistry() {
+        when(discoveryClientMock.getApplication(CoreService.GATEWAY.getServiceId())).thenReturn(GW1_APPLICATION);
+
+        gatewayRegistry.cacheRefreshEventHandler(new CacheRefreshedEvent(), discoveryClientMock);
+
+        assertThat(gatewayRegistry.knownAdditionalGateways.asMap().size(), is(1));
+        assertThat(gatewayRegistry.knownAdditionalGateways.asMap().get(GW1_INSTANCE_ID), is(Arrays.asList(GW1_IP_ADDRESS)));
+
+        assertThat(gatewayRegistry.additionalGatewayIpAddressesReference.get().size(), is(1));
+        assertTrue(gatewayRegistry.additionalGatewayIpAddressesReference.get().containsAll(Arrays.asList(GW1_IP_ADDRESS)));
```

Comment:
Just for the sake of simplification and cleanliness, can you please replace everywhere in this test `containsAll(Arrays.asList` with single element to just `contains`, like below in my suggestion?
```suggestion
        assertTrue(gatewayRegistry.additionalGatewayIpAddressesReference.get().contains(GW1_IP_ADDRESS));
```

Reply:
cleaned up everywhere

Reply:
Thanks!